zipkin-autoconfigure/storage-elasticsearch-http/src/main/java/zipkin/autoconfigure/storage/elasticsearch/http/ZipkinHttpElasticsearchStorageAutoConfiguration.java

```diff
@@ -25,21 +30,61 @@
 import zipkin.autoconfigure.storage.elasticsearch.ZipkinElasticsearchStorageAutoConfiguration;
 import zipkin.storage.StorageComponent;
 
+import static com.google.common.base.Preconditions.checkArgument;
+
 @Configuration
-@EnableConfigurationProperties(ZipkinElasticsearchHttpStorageProperties.class)
+@EnableConfigurationProperties(ZipkinHttpElasticsearchStorageProperties.class)
 @ConditionalOnProperty(name = "zipkin.storage.type", havingValue = "elasticsearch")
-@Conditional(ZipkinHttpElasticsearchStorageAutoConfiguration.HostsAreUrls.class)
+@Conditional({
+    ZipkinHttpElasticsearchStorageAutoConfiguration.HostsAreUrls.class,
+    ZipkinHttpElasticsearchStorageAutoConfiguration.HostsArentAwsUrls.class
+})
 @ConditionalOnMissingBean(StorageComponent.class)
 public class ZipkinHttpElasticsearchStorageAutoConfiguration {
-  @Bean StorageComponent storage(ZipkinElasticsearchHttpStorageProperties elasticsearch) {
+  @Bean StorageComponent storage(ZipkinHttpElasticsearchStorageProperties elasticsearch) {
     return elasticsearch.toBuilder().build();
   }
 
-  public static final class HostsAreUrls implements Condition {
+  static final class HostsAreUrls implements Condition {
     @Override public boolean matches(ConditionContext condition, AnnotatedTypeMetadata md) {
       String hosts = condition.getEnvironment().getProperty("zipkin.storage.elasticsearch.hosts");
       if (hosts == null) return false;
       return ZipkinElasticsearchStorageAutoConfiguration.hostsAreUrls(hosts);
     }
   }
+
+  static final class HostsArentAwsUrls implements Condition {
+    @Override public boolean matches(ConditionContext condition, AnnotatedTypeMetadata md) {
+      String hosts = condition.getEnvironment().getProperty("zipkin.storage.elasticsearch.hosts");
+      if (hosts == null) return true;
+      return !regionFromAwsUrls(Arrays.asList(hosts.split(","))).isPresent();
+    }
+  }
+
+  /**
+   * Only here for consumption by {@link HostsArentAwsUrls}; ideally we could find a way to express
+   * a hierarchy of autoconfigurations in spring without each parent needing to turn themselves off
+   * for their children, but there seems to be no reliable way to order autoconfigure'd classes
+   * (neither @AutoConfigureBefore nor @AutoConfigureAfter seemd to have the desired effect)
+   */
+  private static final Pattern AWS_URL =
+      Pattern.compile("^https://[^.]+\\.([^.]+)\\.es\\.amazonaws\\.com", Pattern.CASE_INSENSITIVE);
+
+  public static Optional<String> regionFromAwsUrls(List<String> hosts) {
```

Comment:
I'm having doubts about maintaining code like this. this sort of stuff has breeds bugs. I'd much rather point people to the existing docs and code like use "cloud.aws.region.auto" or "cloud.aws.region.static", which of course we can map to env variables like AWS_REGION.